helix-core/src/main/java/org/apache/helix/task/WorkflowRebalancer.java

```diff
@@ -496,11 +496,10 @@ private void cleanupJob(final String job, String workflow) {
 
     // Delete job context
     // For recurring workflow, it's OK if the node doesn't exist.
-    String propStoreKey = TaskUtil.getWorkflowContextKey(job);
-    if (!_manager.getHelixPropertyStore().remove(propStoreKey, AccessOption.PERSISTENT)) {
+    if (!TaskUtil.removeJobContext(_manager, job)) {
       LOG.warn(String.format(
           "Error occurred while trying to clean up job %s. Failed to remove node %s from Helix.",
-          job, propStoreKey));
+          job, TaskUtil.getWorkflowContextKey(job)));
```

Comment:
do not need TaskUtil.getWorkflowContextKey(job). just log something like: "Error ... when clean up job's workflow context..."